packages/java-parser/src/unicode.js

```diff
@@ -0,0 +1,144 @@
+"use strict";
+const unicode = {};
+const fs = require("fs");
+const path = require("path");
+const args = process.argv.slice(2)[0];
+let lines;
+try {
+  lines = fs.readFileSync(args, "utf-8").split("\n");
+} catch (err) {
+  throw Error("Please specify the path of the UnicodeData.txt file");
+}
+// The Categories we only want to parse from the file.
+// We don't need to store characters from the UnicodeData file that we are not going to use
+// Only adding characters that are accepted as first character of a Java Identifier. See specs below.
+// https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isJavaIdentifierStart(char)
+const firstIdentCharCategories = new Set([
+  "Ll",
+  "Lm",
+  "Lo",
+  "Lt",
+  "Lu",
+  "Nl",
+  "Sc",
+  "Pc"
+]);
+
+// Adding remaining categories that can be used in a java identifier.
+// https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isJavaIdentifierPart(char)
```

Comment:
Should not we use Java 11 specifications? is there a difference?

Reply:
No there isn't, for some reason java doc 11 has a different pathing in the link and I couldn't find it. I've fixed it now and pasted the link to java 11 docs.